### PR TITLE
Improve documentation for curveToLine function

### PR DESCRIPTION
**Pull request by strk**
Attempt to document semantic of SegmentationToleranceType

### CODE REVIEW

**Comment by strk** [comment]
@m-kuhn can you help decoding the meaning of that MaximumAngle ? It isn't clear to me as the default maximum is 1 degree so it cannot be the angle formed by two consecutive segments...

File: src/core/geometry/qgsabstractgeometry.h [hunk 1]
```diff
@@ -88,6 +88,7 @@ class CORE_EXPORT QgsAbstractGeometry
     //! Segmentation tolerance as maximum angle or maximum difference between approximation and circle
     enum SegmentationToleranceType
     {
+      // Maximum external angle between segments, in radians
```
Comment (m-kuhn) [inline src/core/geometry/qgsabstractgeometry.h]: why not make it a doxygen comment ? `//!< comment`

**Comment by m-kuhn** [comment]
> @m-kuhn can you help decoding the meaning of that MaximumAngle ? It isn't clear to me as the default maximum is 1 degree so it cannot be the angle formed by two consecutive segments...

I think this question should be forwarded to @mhugent . He added this parameter in  https://github.com/strk/QGIS/commit/c0d12dcdeb87a2639479eb99c002aec877c4ec01

**Comment by strk** [comment]
@mhugent Can you help review this please ? I think I got `MaximumAngle` correctly now (after adding tests for it in #4746) but what is `MaximumDifference` ? It does not look like maximum distance between the curve and the line...

**Comment by strk** [comment]
I think "max distance between curve and approximation" is the correct interpretation, as written in ce2f7f6.
Do you confirm @mhugent ?

File: src/core/geometry/qgsabstractgeometry.h [hunk 2]
```diff
@@ -88,7 +88,11 @@ class CORE_EXPORT QgsAbstractGeometry
     //! Segmentation tolerance as maximum angle or maximum difference between approximation and circle
     enum SegmentationToleranceType
     {
+      // Maximum angle between generating radii (lines from arc center
```
Comment (nyalldawson) [inline src/core/geometry/qgsabstractgeometry.h]: Can you make these real doxygen tags so that they appear in the generated api docs?

**Comment by strk** [comment]
Done with a440f93394 but Doxygen 1.8.13 does not show those comments
after `make apidoc`. I dunno what's the problem. Other classes use
that same formatting...

File: src/core/geometry/qgsabstractgeometry.h [hunk 3]
```diff
@@ -88,7 +88,13 @@ class CORE_EXPORT QgsAbstractGeometry
     //! Segmentation tolerance as maximum angle or maximum difference between approximation and circle
     enum SegmentationToleranceType
     {
+
+      /** Maximum angle between generating radii (lines from arc center
+       * to output vertices) */
       MaximumAngle = 0,
```
Comment (nyalldawson) [inline src/core/geometry/qgsabstractgeometry.h]: It's a bit ugly for longer strings like this, but see if changing to

    MaximumAngle = 0, //!< Maximum angle between generating radii (lines from arc center to output vertices)

works for you

Comment (strk) [inline src/core/geometry/qgsabstractgeometry.h]: That other format doesn't change a thing here. Do you see *any* enum ever documented in QGIS apidoc ?

Comment (nyalldawson) [inline src/core/geometry/qgsabstractgeometry.h]: Here's one: https://qgis.org/api/classQgsComposerHtml.html#a21c8432e81a40695647deddea9f1ef23

**Comment by strk** [comment]
Ah ok, so then current formatting also produces it.
It was just me not finding the correct file (you may notice
that the comments are removed from
https://qgis.org/api/qgscomposerhtml_8h_source.html

So, ready for merge, I guess.
doc/api/html/classQgsAbstractGeometry.html does have the members
documentation.

**Comment by strk** [comment]
Merged as 57bf9daaac6245b9f62e8ad01900b046210b381e

**Comment by nyalldawson** [comment]
Perfect - thanks!

**Pull request closed without merging**